airflow/assets/manager.py

```diff
@@ -125,7 +125,9 @@ def register_asset_change(
         # todo: add test so that all usages of internal_api_call are added to rpc endpoint
         asset_model = session.scalar(
             select(AssetModel)
-            .where(AssetModel.uri == asset.uri)
+            .where(
+                AssetModel.name == asset.name, AssetModel.uri == asset.uri, AssetModel.group == asset.group
```

Comment:
`group` is not in the unique key so selecting with it might cause some weird issues in edge cases. It _should not_ matter but…